File: ios/Sources/LocalIpPlugin/LocalIpPlugin.swift
// import Foundation
// import Capacitor

// /**
//  * Please read the Capacitor iOS Plugin Development Guide
//  * here: https://capacitorjs.com/docs/plugins/ios
//  */
// @objc(LocalIpPlugin)
// public class LocalIpPlugin: CAPPlugin, CAPBridgedPlugin {
//     public let identifier = "LocalIpPlugin"
//     public let jsName = "LocalIp"
//     public let pluginMethods: [CAPPluginMethod] = [
//         CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
//     ]
//     private let implementation = LocalIp()

//     @objc func echo(_ call: CAPPluginCall) {
//         let value = call.getString("value") ?? ""
//         call.resolve([
//             "value": implementation.echo(value)
//         ])
//     }
// }

import Foundation
import Capacitor

@objc(LocalIpPlugin)
public class LocalIpPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "LocalIpPlugin"
    public let jsName = "LocalIp"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getLocalIp", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = LocalIp()

    @objc func getLocalIp(_ call: CAPPluginCall) {
        print("LocalIpPlugin.getLocalIp called!")
        if let ip = implementation.getLocalIp() {
            call.resolve([
                "ip": ip
            ])
        } else {
            call.reject("Unable to determine local IP address")
        }
    }
}

File: android/src/main/java/com/cappitolian/plugins/localip/LocalIpPlugin.java
// package com.cappitolian.plugins.localip;

// import com.getcapacitor.JSObject;
// import com.getcapacitor.Plugin;
// import com.getcapacitor.PluginCall;
// import com.getcapacitor.PluginMethod;
// import com.getcapacitor.annotation.CapacitorPlugin;

// @CapacitorPlugin(name = "LocalIp")
// public class LocalIpPlugin extends Plugin {

//     private LocalIp implementation = new LocalIp();

//     @PluginMethod
//     public void echo(PluginCall call) {
//         String value = call.getString("value");

//         JSObject ret = new JSObject();
//         ret.put("value", implementation.echo(value));
//         call.resolve(ret);
//     }
// }

package com.cappitolian.plugins.localip;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.JSObject;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "LocalIp")
public class LocalIpPlugin extends Plugin {
    private LocalIp implementation = new LocalIp();

    @PluginMethod
    public void getLocalIp(PluginCall call) {
        try {
            String localIp = implementation.getLocalIp();
            if (localIp != null) {
                JSObject ret = new JSObject();
                ret.put("ip", localIp);
                call.resolve(ret);
            } else {
                call.reject("Local IP address not found");
            }
        } catch (Exception e) {
            call.reject("Error retrieving IP: " + e.getMessage());
        }
    }
}